utils/runtime-fuzzer/src/generator.rs

```diff
@@ -0,0 +1,259 @@
+
+    const fn claim_value_data_requirement() -> usize {
+        ID_SIZE + AUXILIARY_SIZE
+    }
+}
+
+pub(crate) struct RuntimeStateViewProducer<'a> {
+    corpus_id: String,
+    _unstructured: Unstructured<'a>,
+    sender: AccountId,
+    programs: BTreeSet<ProgramId>,
+    // todo issue - include time limits, so no outdated mailbox messages will be stored.
+    mailbox: BTreeSet<MessageId>,
+}
+
+impl<'a> RuntimeStateViewProducer<'a> {
+    pub(crate) fn new(
+        corpus_id: String,
+        data_requirement: FulfilledDataRequirement<'a, Self>,
+    ) -> Self {
+        Self {
+            corpus_id,
+            _unstructured: Unstructured::new(data_requirement.data),
+            sender: runtime::alice(),
+            programs: BTreeSet::new(),
+            mailbox: BTreeSet::new(),
+        }
+    }
+
+    pub(crate) fn produce_state_view(&mut self) -> RuntimeStateView {
+        self.update_state_view();
+
+        runtime::increase_to_max_balance(self.sender.clone())
+            .unwrap_or_else(|e| unreachable!("Balance update failed: {e:?}"));
+        log::info!(
+            "Current balance of the sender - {}",
+            BalancesPallet::<Runtime>::free_balance(&self.sender)
+        );
+
+        let programs = Vec::from_iter(self.programs.iter());
+        let mailbox = Vec::from_iter(self.mailbox.iter());
+
+        RuntimeStateView {
+            corpus_id: &self.corpus_id,
+            programs,
+            mailbox,
+            max_gas: runtime::default_gas_limit(),
+        }
+    }
+
+    /// Updates mailbox and existing programs view and resets events.
+    fn update_state_view(&mut self) {
+        System::events().iter().for_each(|e| {
+            let RuntimeEvent::Gear(ref gear_event) = e.event else {
+                return;
+            };
+            match gear_event {
+                GearEvent::ProgramChanged {
+                    id,
+                    change: ProgramChangeKind::Active { .. },
+                } => {
+                    self.programs.insert(*id);
+                }
+                GearEvent::UserMessageSent {
+                    message,
+                    expiration: Some(_),
+                } => {
+                    if message.destination() == runtime::alice_program_id() {
+                        self.mailbox.insert(message.id());
+                    }
+                }
+                _ => {}
+            }
+        });
+
+        System::reset_events();
+    }
+}
+
+impl RuntimeStateViewProducer<'_> {
+    pub(crate) const fn random_data_requirement() -> usize {
+        const VALUE_SIZE: usize = mem::size_of::<u128>();
+
+        VALUE_SIZE
+            * (GearCallsGenerator::MAX_UPLOAD_PROGRAM_CALLS
+                + GearCallsGenerator::MAX_SEND_MESSAGE_CALLS
+                + GearCallsGenerator::MAX_SEND_REPLY_CALLS)
+            + AUXILIARY_SIZE
+    }
+}
+
+pub(crate) struct RuntimeStateView<'a> {
+    corpus_id: &'a str,
+    programs: Vec<&'a ProgramId>,
```

Comment:
why not slice?

Reply:
Now it's `Option<&'a NonEmpty<ProgramId>>`. The `RuntimeStateViewProducer` owns `Option<NonEmpty<ProgramId>>`.